soci/index_builder.go

```diff
@@ -134,7 +143,11 @@ func (b *IndexBuilder) BuildIndex(ctx context.Context, desc ocispec.Descriptor)
 		Size:        imgManifestDesc.Size,
 		Annotations: imgManifestDesc.Annotations,
 	}
-	return newOCIArtifactManifest(ztocsDesc, refers, annotations), ztocReaders, nil
+	index := newOCIArtifactManifest(ztocsDesc, refers, annotations)
+	return &IndexWithMetadata{
+		Index:       index,
+		ImageDigest: desc.Digest,
```

Comment:
There was another update recently that added `Platform` to `IndexWithMetadata`. Can you add that here too?

Reply:
yes, will do